src/main/groovy/org/pih/warehouse/importer/ProductSupplierPreferenceImporter.groovy

```diff
@@ -34,8 +34,8 @@ class ProductSupplierPreferenceImporter extends AbstractExcelImporter {
             organizationCode     : ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
             organizationName     : ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
             preferenceTypeName   : ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
-            validityStartDate    : ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
-            validityEndDate      : ([expectedType: ExpectedPropertyType.StringType, defaultValue: null]),
+            validityStartDate    : ([expectedType: ExpectedPropertyType.DateJavaType, defaultValue: null]),
```

Comment:
This should probably just be DateType since we don't need the timestamp component.

Reply:
Unless you can explain why we need the DateJavaType.

Reply:
DataType is LocalDate, so I got it formatted other than 'MM/dd/yyyy', so that's the reason that I have to assign it to params.validityStartDate with format method. Because of that assignment, I got a date displayed in the table with the proper format and also properly imported. I changed it to DateType, but I am still using the format method.
By the "DataType is LocalDate" I mean the fact that my imported Date looks like this (at the left the date is using the format method, at the right is just DateType):
![image](https://github.com/openboxes/openboxes/assets/83239466/fe3c27d3-22c1-47cd-a8a5-bcbbf97df48f)

Because of using DateType / DateJavaType in our import, wrong data is not visible in the table to import, so it can be imported - it can lead to entering wrong data, with some missing dates.
By this, I mean:
- My Excel looks like this:
![image](https://github.com/openboxes/openboxes/assets/83239466/9b82e067-ee79-4e38-b600-f74726603f04)
- Dates to import look like this (I didn't see any validation)
![image](https://github.com/openboxes/openboxes/assets/83239466/ac28148f-e77f-454d-a6c4-0da143ab08b7)
Let me know what you think about this behavior of importing with `ExpectedPropertyType.DateType` - maybe it can be the one argument for using our own date parser

For now, I am pushing the change with DateJavaType -> DateType and with getting rid of:
def dateFormat = new SimpleDateFormat("MM/dd/yyyy")
if (validityStartDate) {
    try {
        dateFormat.parse(validityStartDate)
    } catch (Exception e) {
        command.errors.reject("Row ${index + 1}: Validity start date ${validityStartDate} is invalid")
    }
}

Reply:
The PR looks good. However, you're right ... the silent failure on invalid dates doesn't feel right. I'll let @awalkowiak make a final decision on whether we're ok merging as-is or whether we want to do deal with that in this PR. 

The only way I can think of doing that is turning the data type back to String and doing the parsing / validation ourselves again. Otherwise, if the importer plugin just returns null for bad data, then we won't have a reference to the bad data and we'll never know if a value was entered into the column.

One other way that might not yet be explored is if there's some way to configure the importer to warn us when it gets invalid data. That would be the best solution.

Reply:
Alright, I did a little digging and found that there's a way to do this through the plugin infrastructure. 
```
    static cellReporter = new DefaultImportCellCollector()

    List<Map> getData() {
        return Holders.grailsApplication.mainContext.getBean("excelImportService")
                .convertColumnMapConfigManyRows(workbook, columnMap, cellReporter, null, propertyMap)
    }
```
I'm not sure if static is correct here but that's what I saw with the excel import plugin test code.
https://github.com/gpc/grails-excel-import/blob/master/src/test/groovy/sample/BookImporter.groovy

But that might have been done as a convenience for testing purposes.

then we can get error messages using the cell reporter (after the data has been uploaded)
```
dataImporter.cellReporter.messagesBySeverityLevel
```

Unfortunately, there's an error when the DefaultImportCellCollector insteance tries to report an error for a cell with JavaDateType property type. The default cell reporter doesn't seem to know how to handle this data type.
```
2023-08-20 09:29:03,527 ERROR [080-exec-7] o.g.p.e.DefaultImportCellCollector      : Exception while trying to report potential problem
groovy.lang.MissingMethodException: No signature of method: org.grails.plugins.excelimport.ImportSeverityLevelEnum.severityLevel() is applicable for argument types: (java.lang.Integer, org.grails.plugins.excelimport.ExpectedPropertyType) values: [1, DateJavaType]
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:71)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:53)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.grails.plugins.excelimport.ImportSeverityMappingEnum$severityLevel.call(Unknown Source)
        at 
```
So I think we just need to override the default cell reporter or create our own CellReporter class to log these issues and prevent the import from proceeding if there are errors.

My two cents: This is a refactoring and should be a separate ticket. 

And instead of putting the decision on @awalkowiak to merge or not merge, it'll probably be best to ask Kelsey and Manon if they are ok with the silent fail behavior here. If not then we'll need to implement this custom error reporter mechanism, so this ticket becomes at least a 5.

Reply:
And for now, if Kelsey and Manon *do* want to prevent bad date data, then we can go back to handling the date as a StringType or DoubleType.

https://stackoverflow.com/a/19028309/136597

```
import org.apache.poi.ss.usermodel.DateUtil

Date validityStartDate
if (params.validityStartDate) {
    validityStartDate = DateUtil.getJavaDate(params.validityStartDate)
}
```